openedx/core/djangoapps/oauth_dispatch/migrations/0013_add_index_to_token_tables.py

```diff
@@ -0,0 +1,25 @@
+# Generated by Django 3.2.18 on 2023-03-27 12:39
+
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('oauth_dispatch', '0012_noop_migration_to_test_rollback'),
+    ]
+
+    operations = [
+        # CREATE INDEX is a non-blocking operation in MySql 5.6 and above
+        # Adding as raw SQL because we don't have access to the underlying model.
+        #
+        # WARNING:
+        # This migration may cause problems in the future if django-oauth-toolkit ever decides to update the AccessToken
+        # model with the same index name. However, if they do it using the ORM and autogenerated migrations it will
+        # add a hash onto the index name so we should be safe
+        migrations.RunSQL(
+            sql='''CREATE INDEX oauth2_provider_accesstoken_expires ON oauth2_provider_accesstoken (expires);''',
```

Comment:
Based on these:
- https://dev.mysql.com/doc/refman/5.7/en/create-index.html#create-index-copying
- https://dev.mysql.com/doc/refman/5.7/en/alter-table.html#alter-table-concurrency
- https://dev.mysql.com/doc/refman/5.7/en/innodb-online-ddl.html
- https://dev.mysql.com/doc/refman/5.7/en/alter-table.html#alter-table-performance

I believe this will cause the operation to throw up its hands instead of trying to create the index and blocking the table if it can't be performed in a non-blocking way.
```suggestion
            sql='''CREATE INDEX oauth2_provider_accesstoken_expires ON oauth2_provider_accesstoken (expires), ALGORITHM=INPLACE, LOCK=NONE;''',
```

Reply:
Ah. I misunderstood what I read in the tables. I thought by "supports in place" and the like they meant that was the default.